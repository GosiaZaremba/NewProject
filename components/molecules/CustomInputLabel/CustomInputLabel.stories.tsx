import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomInputLabel } from './CustomInputLabel';
import { boolean, text } from '@storybook/addon-knobs';

storiesOf('molecules/CustomInputLabel', module)
    .addDecorator((story: () => void) => story())
    .add('default', () => (
        <CustomInputLabel
            labelText={text('Label Text', 'Label Text')}
            placeholder={text('Placeholder', 'Placeholder')}
            secureTextEntry={boolean('secureTextEntry', false)}
        />
    ));
