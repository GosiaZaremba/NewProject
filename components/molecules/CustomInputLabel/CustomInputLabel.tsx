import React, { forwardRef, useImperativeHandle, useRef } from 'react';
import { CustomInput, CustomInputReference, Label } from '../../atoms';
import { View } from 'react-native';

export type Props = {
    labelText: string;
    placeholder: string;
    secureTextEntry: boolean;
    testID?: string;
};

export type CustomInputLabelReference = {
    getValue: () => string;
    focus: () => void;
};

export const CustomInputLabelWithReference: React.ForwardRefRenderFunction<
    CustomInputLabelReference,
    Props
> = ({ labelText, placeholder, secureTextEntry, testID }, ref) => {
    const inputReference = useRef<CustomInputReference>(null);

    useImperativeHandle(ref, () => ({
        getValue: () => {
            return inputReference.current?.getValue() || '';
        },
        focus: () => {
            inputReference?.current?.focus();
        },
    }));

    return (
        <View testID={testID}>
            <Label text={labelText} />
            <CustomInput
                placeholder={placeholder}
                secureTextEntry={secureTextEntry}
                ref={inputReference}
            />
        </View>
    );
};

export const CustomInputLabel = forwardRef(CustomInputLabelWithReference);
