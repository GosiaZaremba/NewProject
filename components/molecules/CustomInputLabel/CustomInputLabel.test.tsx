import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomInputLabel } from './CustomInputLabel';

describe('Molecule InputLabel', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomInputLabel
                    placeholder={'Custom InputLabel'}
                    secureTextEntry={false}
                    testID={'molecule-inputLabel'}
                    labelText={'LabelText'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <CustomInputLabel
                placeholder={'Custom InputLabel'}
                secureTextEntry={false}
                testID={'molecule-inputLabel'}
                labelText={'LabelText'}
            />
        );
        const customInputLabel = getByTestId('molecule-inputLabel');
        // fireEvent.changeText(customInputLabel, '123');
        // expect(customInputLabel.props.value).toBe('123');
        // expect(customInputLabel.props.placeholder).toEqual('Custom InputLabel');
        // expect(customInputLabel.props.secureTextEntry).toEqual(false);
        // expect(customInput.props.onChangeText()).toHaveReturned();
        // console.log(customInputLabel.props);
    });
});
