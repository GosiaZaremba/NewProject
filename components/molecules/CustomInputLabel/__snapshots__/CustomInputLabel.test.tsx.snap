// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule InputLabel Component Snapshot 1`] = `
<View
  testID="molecule-inputLabel"
>
  <Text
    style={
      Object {
        "color": "black",
        "marginBottom": 2,
      }
    }
  >
    LabelText
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Custom InputLabel"
    secureTextEntry={false}
    style={
      Object {
        "borderColor": "#908D8E",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 10,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;
